tests/functional/test_stub.py

```diff
@@ -0,0 +1,162 @@
+# Copyright 2014 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+from tests import unittest
+import mock
+
+import botocore
+from botocore.stub import QueueStubber
+from botocore.credentials import Credentials
+from botocore.exceptions import StubResponseError
+from botocore import hooks
+import botocore.client
+import botocore.retryhandler
+import botocore.translate
+
+
+class TestStubber(unittest.TestCase):
```

Comment:
One of the things I was hoping to see in the functional tests was trying to stub out an existing real AWS service to ensure that everything works with a real service model.

This is a final test to ensure that all the pieces fit together and won't be subject to false test passes if the underlying APIs being mocked are changed.